target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8068-ap3935.dts

```diff
@@ -295,55 +310,51 @@
 	};
 };
 
-&soc {
-	mdio1: mdio {
-		compatible = "virtual,mdio-gpio";
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		status = "okay";
+&mdio0 {
+	status = "okay";
 
-		pinctrl-0 = <&mdio0_pins>;
-		pinctrl-names = "default";
+	pinctrl-0 = <&mdio0_pins>;
+	pinctrl-names = "default";
 
-		gpios = <&qcom_pinmux 1 GPIO_ACTIVE_HIGH &qcom_pinmux 0 GPIO_ACTIVE_HIGH>;
+	gpios = <&qcom_pinmux 1 GPIO_ACTIVE_HIGH &qcom_pinmux 0 GPIO_ACTIVE_HIGH>;
 
-		phy1: ethernet-phy@1 {
-			reg = <1>;
-		};
+	phy1: ethernet-phy@1 {
+		reg = <1>;
+	};
 
-		phy2: ethernet-phy@2 {
-			reg = <2>;
-		};
+	phy2: ethernet-phy@2 {
+		reg = <2>;
 	};
 };
 
 &gmac0 {
 	status = "okay";
 
 	qcom,id = <0>;
-	mdiobus = <&mdio1>;
+	mdiobus = <&mdio0>;
+
+	pinctrl-0 = <&rgmii0_pins>;
+	pinctrl-names = "default";
 
-	phy-mode = "rgmii";
+	phy-mode = "rgmii-id";
 	phy-handle = <&phy1>;
 
 	nvmem-cells = <&ethaddr>;
 	nvmem-cell-names = "mac-address";
-
-	fixed-link {
-		speed = <1000>;
-		full-duplex;
-	};
 };
 
 &gmac2 {
 	status = "okay";
 
 	qcom,id = <2>;
-	mdiobus = <&mdio1>;
+	mdiobus = <&mdio0>;
 
 	phy-mode = "sgmii";
 	phy-handle = <&phy2>;
+
+	nvmem-cells = <&ethaddr>;
+	nvmem-cell-names = "mac-address";
+	mac-address-increment = <1>;
```

Comment:
Deprecated please use nvmem-layout new way with nvmem cell cells. (you can check other example in other dts)